server/src/main/java/org/opensearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -398,6 +399,46 @@ protected Object rewriteForDocValue(Object value) {
             return value;
         }
 
+        @Override
+        public Query termQueryCaseInsensitive(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            if (isSearchable()) {
+                return super.termQueryCaseInsensitive(value, context);
+            } else {
+                BytesRef bytesRef = indexedValueForSearch(rewriteForDocValue(value));
+                Term term = new Term(name(), bytesRef);
+                Query query = AutomatonQueries.createAutomatonQuery(
+                    term,
+                    AutomatonQueries.toCaseInsensitiveString(bytesRef.utf8ToString(), Integer.MAX_VALUE),
```

Comment:
Instead of `Integer.MAX_VALUE`, can we please make this `Operations.DEFAULT_DETERMINIZE_WORK_LIMIT` for now?

I'm about to open an issue to address this across all case-insensitive query uses (see my comment at https://github.com/opensearch-project/documentation-website/issues/9028#issuecomment-2575913081).